src/visuals-api.d.ts

```diff
@@ -0,0 +1,1788 @@
+        Web = 1 << 0,
+        PublishToWeb = 1 << 1,
+        Desktop = 1 << 2,
+        Embed = 1 << 3,
+        ReportServer = 1 << 4,
+        ExportReportHost = 1 << 5,
+        Mobile = 1 << 6
+    }
+}
+
+declare module powerbi {
+    export interface IFilter { }
+}
+
+/**
+ * Change Log Version 1.13.0
+ *  Expanded `host.colorPalette` now expose a boolean `isHighContrast` flag and several non-data colors 
+ *  including `foreground`, `foregroundSelected`, `background` and `hyperlink` all of which are required for high-contrast accessibility support.
+ */
+
+declare module powerbi.extensibility.visual {
+    /**
+     * Represents a visualization displayed within an application (PowerBI dashboards, ad-hoc reporting, etc.).
+     * This interface does not make assumptions about the underlying JS/HTML constructs the visual uses to render itself.
+     */
+    export interface IVisual extends extensibility.IVisual {
+        /** Notifies the IVisual of an update (data, viewmode, size change). */
+        update<T>(options: VisualUpdateOptions, viewModel?: T): void;
+
+        /** Notifies the visual that it is being destroyed, and to do any cleanup necessary (such as unsubscribing event handlers). */
+        destroy?(): void;
+
+        /** Gets the settings to display in the formatting pane */
+        getFormattingModel?(): visuals.FormattingModel | undefined;
+    }
+
+    export interface IVisualHost extends extensibility.IVisualHost {
+        createSelectionIdBuilder: () => visuals.ISelectionIdBuilder;
+        createSelectionManager: () => ISelectionManager;
+        colorPalette: ISandboxExtendedColorPalette;
+        persistProperties: (changes: VisualObjectInstancesToPersist) => void;
+        applyJsonFilter: (filter: IFilter[] | IFilter, objectName: string, propertyName: string, action: FilterAction) => void;
+        tooltipService: ITooltipService;
+        telemetry: ITelemetryService;
+        authenticationService: IAuthenticationService;
+        locale: string;
+        hostCapabilities: HostCapabilities;
+        launchUrl: (url: string) => void;
+        fetchMoreData: (aggregateSegments?: boolean) => boolean;
+        openModalDialog: (dialogId: string, options?: DialogOpenOptions, initialState?: object) => IPromise<ModalDialogResult>;
+        instanceId: string;
+        refreshHostData: () => void;
+        createLocalizationManager: () => ILocalizationManager;
+        storageService: ILocalVisualStorageService;
+        downloadService: IDownloadService;
+        eventService: IVisualEventService;
+        switchFocusModeState: (on: boolean) => void;
+        hostEnv: powerbi.common.CustomVisualHostEnv;
+        displayWarningIcon: (hoverText: string, detailedText: string) => void;
+        licenseManager: IVisualLicenseManager;
+        webAccessService: IWebAccessService;
+        drill: (args: DrillArgs) => void;
+        applyCustomSort: (args: CustomVisualApplyCustomSortArgs) => void;
```

Comment:
Does this give the opportunity to run query code in our side PBIClients (outside of the iframe), cause if yes it may be a security issue

Reply:
@dbrailov ?